From 42a1de2b78ca5f07032743c9e7c88c03fdf4e03e Mon Sep 17 00:00:00 2001
From: Maxim Orlov <orlovmg@gmail.com>
Date: Mon, 24 Nov 2025 19:18:10 +0300
Subject: [PATCH v67 12/12] Fix MSVC build

Fix guc_tables.h(303): error C2628: 'config_int64' followed by
    '__int64' is illegal (did you forget a ';'?)
---
 src/backend/utils/misc/gen_guc_tables.pl  |  2 +-
 src/backend/utils/misc/guc.c              | 62 +++++++++++------------
 src/backend/utils/misc/guc_funcs.c        |  2 +-
 src/backend/utils/misc/guc_parameters.dat | 16 +++---
 src/backend/utils/misc/guc_tables.c       |  2 +-
 src/backend/utils/misc/help_config.c      |  8 +--
 src/include/utils/guc_tables.h            |  4 +-
 7 files changed, 48 insertions(+), 48 deletions(-)

diff --git a/src/backend/utils/misc/gen_guc_tables.pl b/src/backend/utils/misc/gen_guc_tables.pl
index 32ae3536da..48a6a25425 100644
--- a/src/backend/utils/misc/gen_guc_tables.pl
+++ b/src/backend/utils/misc/gen_guc_tables.pl
@@ -72,7 +72,7 @@ sub print_table
 		printf $ofh "\t\t._%s = {\n", $entry->{type};
 		printf $ofh "\t\t\t.variable = &%s,\n", $entry->{variable};
 
-		if ($entry->{type} eq 'int64')
+		if ($entry->{type} eq 'int_64')
 		{
 			printf $ofh "\t\t\t.boot_val = INT64CONST(%s),\n", $entry->{boot_val};
 			printf $ofh "\t\t\t.min = INT64CONST(%s),\n", $entry->{min};
diff --git a/src/backend/utils/misc/guc.c b/src/backend/utils/misc/guc.c
index c1be5d1a2d..9aa16a2f36 100644
--- a/src/backend/utils/misc/guc.c
+++ b/src/backend/utils/misc/guc.c
@@ -794,8 +794,8 @@ set_stack_value(struct config_generic *gconf, config_var_value *val)
 		case PGC_INT:
 			val->val.intval = *gconf->_int.variable;
 			break;
-		case PGC_INT64:
-			val->val.int64val = *gconf->_int64.variable;
+		case PGC_INT_64:
+			val->val.int64val = *gconf->_int_64.variable;
 			break;
 		case PGC_REAL:
 			val->val.realval = *gconf->_real.variable;
@@ -821,7 +821,7 @@ discard_stack_value(struct config_generic *gconf, config_var_value *val)
 	{
 		case PGC_BOOL:
 		case PGC_INT:
-		case PGC_INT64:
+		case PGC_INT_64:
 		case PGC_REAL:
 		case PGC_ENUM:
 			/* no need to do anything */
@@ -1346,13 +1346,13 @@ check_GUC_init(const struct config_generic *gconf)
 				}
 				break;
 			}
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				const struct config_int64 *conf = &gconf->_int64;
+				const struct config_int64 *conf = &gconf->_int_64;
 
 				if (*conf->variable != 0 && *conf->variable != conf->boot_val)
 				{
-					elog(LOG, "GUC (PGC_INT64) %s, boot_val=%" PRIi64 ", C-var=%" PRIi64,
+					elog(LOG, "GUC (PGC_INT_64) %s, boot_val=%" PRIi64 ", C-var=%" PRIi64,
 						 gconf->name, conf->boot_val, *conf->variable);
 					return false;
 				}
@@ -1586,9 +1586,9 @@ InitializeOneGUCOption(struct config_generic *gconf)
 				*conf->variable = conf->reset_val = newval;
 				break;
 			}
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				struct config_int64 *conf = &gconf->_int64;
+				struct config_int64 *conf = &gconf->_int_64;
 				int64		newval = conf->boot_val;
 
 				Assert(newval >= conf->min);
@@ -1960,9 +1960,9 @@ ResetAllOptions(void)
 									gconf->reset_extra);
 					break;
 				}
-			case PGC_INT64:
+			case PGC_INT_64:
 				{
-					struct config_int64 *conf = &gconf->_int64;
+					struct config_int64 *conf = &gconf->_int_64;
 
 					if (conf->assign_hook)
 						conf->assign_hook(conf->reset_val,
@@ -2359,9 +2359,9 @@ AtEOXact_GUC(bool isCommit, int nestLevel)
 							}
 							break;
 						}
-					case PGC_INT64:
+					case PGC_INT_64:
 						{
-							struct config_int64 *conf = &gconf->_int64;
+							struct config_int64 *conf = &gconf->_int_64;
 							int64		newval = newvalue.val.intval;
 							void	   *newextra = newvalue.extra;
 
@@ -3187,9 +3187,9 @@ parse_and_validate_value(const struct config_generic *record,
 					return false;
 			}
 			break;
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				const struct config_int64 *conf = &record->_int64;
+				const struct config_int64 *conf = &record->_int_64;
 				const char *hintmsg;
 
 				if (!parse_int64(value, &newval->int64val,
@@ -3928,9 +3928,9 @@ set_config_with_handle(const char *name, config_handle *handle,
 #undef newval
 			}
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				struct config_int64 *conf = &record->_int64;
+				struct config_int64 *conf = &record->_int_64;
 
 #define newval (newval_union.int64val)
 
@@ -4498,7 +4498,7 @@ GetConfigOption(const char *name, bool missing_ok, bool restrict_privileged)
 					 *record->_int.variable);
 			return buffer;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			snprintf(buffer, sizeof(buffer), "%" PRId64,
 					 *((struct config_int64 *) record)->variable);
 			return buffer;
@@ -4551,7 +4551,7 @@ GetConfigOptionResetString(const char *name)
 					 record->_int.reset_val);
 			return buffer;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			snprintf(buffer, sizeof(buffer), "%" PRId64,
 					 ((struct config_int64 *) record)->reset_val);
 			return buffer;
@@ -5512,9 +5512,9 @@ get_explain_guc_options(int *num)
 				}
 				break;
 
-			case PGC_INT64:
+			case PGC_INT_64:
 				{
-					struct config_int64 *lconf = &conf->_int64;
+					struct config_int64 *lconf = &conf->_int_64;
 
 					modified = (lconf->boot_val != *(lconf->variable));
 				}
@@ -5652,9 +5652,9 @@ ShowGUCOption(const struct config_generic *record, bool use_units)
 			}
 			break;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				const struct config_int64 *conf = &record->_int64;
+				const struct config_int64 *conf = &record->_int_64;
 
 				if (conf->show_hook)
 					val = conf->show_hook();
@@ -5783,9 +5783,9 @@ write_one_nondefault_variable(FILE *fp, struct config_generic *gconf)
 			}
 			break;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				struct config_int64 *conf = &gconf->_int64;
+				struct config_int64 *conf = &gconf->_int_64;
 
 				fprintf(fp, "%" PRId64, *conf->variable);
 			}
@@ -6063,9 +6063,9 @@ estimate_variable_size(struct config_generic *gconf)
 			}
 			break;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				struct config_int64 *conf = &gconf->_int64;
+				struct config_int64 *conf = &gconf->_int_64;
 
 				/*
 				 * Instead of getting the exact display length, use max
@@ -6247,9 +6247,9 @@ serialize_variable(char **destptr, Size *maxbytes,
 			}
 			break;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
-				struct config_int64 *conf = &gconf->_int64;
+				struct config_int64 *conf = &gconf->_int_64;
 
 				do_serialize(destptr, maxbytes, "%" PRId64, *conf->variable);
 			}
@@ -6458,7 +6458,7 @@ RestoreGUCState(void *gucstate)
 		{
 			case PGC_BOOL:
 			case PGC_INT:
-			case PGC_INT64:
+			case PGC_INT_64:
 			case PGC_REAL:
 			case PGC_ENUM:
 				/* no need to do anything */
@@ -7048,7 +7048,7 @@ call_int64_check_hook(const struct config_generic *conf, int64 *newval, void **e
 					GucSource source, int elevel)
 {
 	/* Quick success if no hook */
-	if (!conf->_int64.check_hook)
+	if (!conf->_int_64.check_hook)
 		return true;
 
 	/* Reset variables that might be set by hook */
@@ -7057,7 +7057,7 @@ call_int64_check_hook(const struct config_generic *conf, int64 *newval, void **e
 	GUC_check_errdetail_string = NULL;
 	GUC_check_errhint_string = NULL;
 
-	if (!conf->_int64.check_hook(newval, extra, source))
+	if (!conf->_int_64.check_hook(newval, extra, source))
 	{
 		ereport(elevel,
 				(errcode(GUC_check_errcode_value),
diff --git a/src/backend/utils/misc/guc_funcs.c b/src/backend/utils/misc/guc_funcs.c
index 63a8672834..910ac6b607 100644
--- a/src/backend/utils/misc/guc_funcs.c
+++ b/src/backend/utils/misc/guc_funcs.c
@@ -699,7 +699,7 @@ GetConfigOptionValues(const struct config_generic *conf, const char **values)
 			}
 			break;
 
-		case PGC_INT64:
+		case PGC_INT_64:
 			{
 				struct config_int64 *lconf = (struct config_int64 *) conf;
 
diff --git a/src/backend/utils/misc/guc_parameters.dat b/src/backend/utils/misc/guc_parameters.dat
index 5fb584bf92..a3fa8a9c83 100644
--- a/src/backend/utils/misc/guc_parameters.dat
+++ b/src/backend/utils/misc/guc_parameters.dat
@@ -146,7 +146,7 @@
 
 # see varsup.c for why this is PGC_POSTMASTER not PGC_SIGHUP
 # see vacuum_failsafe_age if you change the upper-limit value.
-{ name => 'autovacuum_freeze_max_age', type => 'int64', context => 'PGC_POSTMASTER', group => 'VACUUM_AUTOVACUUM',
+{ name => 'autovacuum_freeze_max_age', type => 'int_64', context => 'PGC_POSTMASTER', group => 'VACUUM_AUTOVACUUM',
   short_desc => 'Age at which to autovacuum a table to prevent transaction ID wraparound.',
   variable => 'autovacuum_freeze_max_age',
   boot_val => '200000000',
@@ -163,7 +163,7 @@
 },
 
 # see multixact.c for why this is PGC_POSTMASTER not PGC_SIGHUP
-{ name => 'autovacuum_multixact_freeze_max_age', type => 'int64', context => 'PGC_POSTMASTER', group => 'VACUUM_AUTOVACUUM',
+{ name => 'autovacuum_multixact_freeze_max_age', type => 'int_64', context => 'PGC_POSTMASTER', group => 'VACUUM_AUTOVACUUM',
   short_desc => 'Multixact age at which to autovacuum a table to prevent multixact wraparound.',
   variable => 'autovacuum_multixact_freeze_max_age',
   boot_val => '400000000',
@@ -3219,7 +3219,7 @@
   max => '10000',
 },
 
-{ name => 'vacuum_failsafe_age', type => 'int64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
+{ name => 'vacuum_failsafe_age', type => 'int_64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
   short_desc => 'Age at which VACUUM should trigger failsafe to avoid a wraparound outage.',
   variable => 'vacuum_failsafe_age',
   boot_val => '1600000000',
@@ -3227,7 +3227,7 @@
   max => '2100000000',
 },
 
-{ name => 'vacuum_freeze_min_age', type => 'int64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
+{ name => 'vacuum_freeze_min_age', type => 'int_64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
   short_desc => 'Minimum age at which VACUUM should freeze a table row.',
   variable => 'vacuum_freeze_min_age',
   boot_val => '50000000',
@@ -3235,7 +3235,7 @@
   max => '0x7FFFFFFFFFFFFFFF',
 },
 
-{ name => 'vacuum_freeze_table_age', type => 'int64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
+{ name => 'vacuum_freeze_table_age', type => 'int_64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
   short_desc => 'Age at which VACUUM should scan whole table to freeze tuples.',
   variable => 'vacuum_freeze_table_age',
   boot_val => '150000000',
@@ -3252,7 +3252,7 @@
   max => '1.0',
 },
 
-{ name => 'vacuum_multixact_failsafe_age', type => 'int64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
+{ name => 'vacuum_multixact_failsafe_age', type => 'int_64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
   short_desc => 'Multixact age at which VACUUM should trigger failsafe to avoid a wraparound outage.',
   variable => 'vacuum_multixact_failsafe_age',
   boot_val => '1600000000',
@@ -3260,7 +3260,7 @@
   max => '2100000000',
 },
 
-{ name => 'vacuum_multixact_freeze_min_age', type => 'int64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
+{ name => 'vacuum_multixact_freeze_min_age', type => 'int_64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
   short_desc => 'Minimum age at which VACUUM should freeze a MultiXactId in a table row.',
   variable => 'vacuum_multixact_freeze_min_age',
   boot_val => '5000000',
@@ -3268,7 +3268,7 @@
   max => '0x7FFFFFFFFFFFFFFF',
 },
 
-{ name => 'vacuum_multixact_freeze_table_age', type => 'int64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
+{ name => 'vacuum_multixact_freeze_table_age', type => 'int_64', context => 'PGC_USERSET', group => 'VACUUM_FREEZING',
   short_desc => 'Multixact age at which VACUUM should scan whole table to freeze tuples.',
   variable => 'vacuum_multixact_freeze_table_age',
   boot_val => '150000000',
diff --git a/src/backend/utils/misc/guc_tables.c b/src/backend/utils/misc/guc_tables.c
index 5d8feaf3ac..8656ffb0af 100644
--- a/src/backend/utils/misc/guc_tables.c
+++ b/src/backend/utils/misc/guc_tables.c
@@ -752,7 +752,7 @@ const char *const config_type_names[] =
 {
 	[PGC_BOOL] = "bool",
 	[PGC_INT] = "integer",
-	[PGC_INT64] = "int64",
+	[PGC_INT_64] = "int64",
 	[PGC_REAL] = "real",
 	[PGC_STRING] = "string",
 	[PGC_ENUM] = "enum",
diff --git a/src/backend/utils/misc/help_config.c b/src/backend/utils/misc/help_config.c
index 5296f50d23..acbe0263c7 100644
--- a/src/backend/utils/misc/help_config.c
+++ b/src/backend/utils/misc/help_config.c
@@ -90,11 +90,11 @@ printMixedStruct(const struct config_generic *structToPrint)
 				   structToPrint->_int.min,
 				   structToPrint->_int.max);
 			break;
-		case PGC_INT64:
+		case PGC_INT_64:
 			printf("INTEGER64\t%" PRIi64 "\t%" PRIi64 "\t%" PRIi64 "\t",
-				   structToPrint->_int64.reset_val,
-				   structToPrint->_int64.min,
-				   structToPrint->_int64.max);
+				   structToPrint->_int_64.reset_val,
+				   structToPrint->_int_64.min,
+				   structToPrint->_int_64.max);
 			break;
 		case PGC_REAL:
 			printf("REAL\t%g\t%g\t%g\t",
diff --git a/src/include/utils/guc_tables.h b/src/include/utils/guc_tables.h
index 0865452a85..5406a22be2 100644
--- a/src/include/utils/guc_tables.h
+++ b/src/include/utils/guc_tables.h
@@ -24,7 +24,7 @@ enum config_type
 {
 	PGC_BOOL,
 	PGC_INT,
-	PGC_INT64,
+	PGC_INT_64,
 	PGC_REAL,
 	PGC_STRING,
 	PGC_ENUM,
@@ -300,7 +300,7 @@ struct config_generic
 	{
 		struct config_bool _bool;
 		struct config_int _int;
-		struct config_int64 _int64;
+		struct config_int64 _int_64;	/* MSVC bail out on _int64 */
 		struct config_real _real;
 		struct config_string _string;
 		struct config_enum _enum;
-- 
2.51.0

